### PR TITLE
feat: convert RWD cert projects

### PR DESCRIPTION
**Pull request by moT01**
Not sure if we want to wait on other things for this or just go with it. It converts the 10 english RWD cert projects to use the multifile editor. It also adds some cypress tests for saving and loading code, and fixes some small inconsistencies in variable names. The commit messages are pretty descriptive.

ToDo:
~~Add solutions~~
Add to i18n's?

~~Look into potential issue about running failing tests and that saves code to local storage and then getting loaded over what's in database or something.~~

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Related to https://github.com/freeCodeCamp/freeCodeCamp/issues/44664

### CODE REVIEW

**Comment by ShaunSHamilton** [comment]
I highly suggest we make use of the `--fcc-actual--` and `--fcc-expected--`

**Comment by moT01** [comment]
I'm in favor of that @ShaunSHamilton - Although, I'm not sure how it would look with these. The current test suite allows multiple assert statements per test. So you can check and give feedback for specific things within a test. For example, the user story - ``For the name, email, and number input fields, you can see corresponding `label` elements in the form, that describe the purpose of each field with the following ids: `id="name-label"`, `id="email-label"`, and `id="number-label"` `` requires multiple things to check - the current suite would be able to have specific feedback for each thing checked - I don't think we can do that at the moment with what we have here. Certainly, we could figure out something if we wanted. Perhaps this can be good enough to start.

**Comment by ShaunSHamilton** [comment]
> Although, I'm not sure how it would look with these. The current test suite allows multiple assert statements per test. So you can check and give feedback for specific things within a test.

I was moreso thinking we make the tests align with the current test-suite, and not the user-stories. That is, have 4x more tests than user stories so we can give specific feedback.

**Comment by moT01** [comment]
Yea, that could work. Perhaps we should have a discussion on it. There was a little about it somewhere, where we (Nick and I) decided to just put the user stories in the as the test text, but maybe that's not the way to go.

**Comment by ShaunSHamilton** [comment]
@moT01 Well, I would do that too.

The description should contain the user-stories as they are now (potentially with one requiring the DOCTYPE), but the tests/hints should contain as much as the testable-repo does.

**Comment by nhcarrigan** [comment]
I'm in agreement that we can have as many tests as we want - as long we aren't testing for additional things that would break backwards compatibility.

File: curriculum/challenges/english/14-responsive-web-design-22/build-a-personal-portfolio-webpage-project/build-a-personal-portfolio-webpage.md [hunk 1]
```diff
@@ -135,6 +135,7 @@ assert.isAtLeast(new __helpers.CSSHelp(document).getCSSRules('media')?.length, 1
 The height of your `#welcome-section` element should equal the height of the viewport.
 
 ```js
+console.log(window.innerHeight)
```
Comment (ShaunSHamilton) [inline curriculum/challenges/english/14-responsive-web-design-22/build-a-personal-portfolio-webpage-project/build-a-personal-portfolio-webpage.md:138]: This worked for me?

Comment (moT01) [reply curriculum/challenges/english/14-responsive-web-design-22/build-a-personal-portfolio-webpage-project/build-a-personal-portfolio-webpage.md:138]: This particular one may have given the correct value, but I think the window is referring to one of the other panes or something. When I was checking the `innerWidth` I was getting like 215, so it seemed like it was the instruction pane or one of the code panes - not sure.

**Comment by ShaunSHamilton** [comment]
We should add solutions for these, no?

**Comment by moT01** [comment]
Not sure if we want to put solutions to cert projects in the code base

**Comment by ShaunSHamilton** [comment]
> Not sure if we want to put solutions to cert projects in the code base

We do for the JS section, for this reason.

**Comment by nhcarrigan** [comment]
The academic honesty policy still applies, and with the moderation queue coming soon:tm: I think it's less of a concern to have the code visible.

File: curriculum/challenges/_meta/build-a-personal-portfolio-webpage-project/meta.json [hunk 2]
```diff
@@ -2,7 +2,7 @@
   "name": "Build a Personal Portfolio Webpage Project",
   "isUpcomingChange": false,
   "usesMultiFileEditor": true,
-  "order": 19,
+  "order": 0,
```
Comment (ShaunSHamilton) [inline curriculum/challenges/_meta/build-a-personal-portfolio-webpage-project/meta.json:5]: Changing these to `0` was necessary to pass the tests. 🤷‍♂️

**Comment by ShaunSHamilton** [comment]
Something I am not sure we entirely thought of is the tests that break when the screen is too big/small.

I assume these tests will always fail in CI, and are impossible to pass on mobile.

**Comment by nhcarrigan** [comment]
> Something I am not sure we entirely thought of is the tests that break when the screen is too big/small.

We shouldn't be testing any rendered content though? We should be looking at the actual CSS with that fancy helper that someone kindly built which even handles `@media` queries.

**Comment by ShaunSHamilton** [comment]
> We shouldn't be testing any rendered content though? We should be looking at the actual CSS with that fancy helper that someone kindly built which even handles `@media` queries.

Right. The issue is how many dozens of ways there are to:
> Your `#nav-bar` should always be at the top of the viewport

> Your `#navbar` should always be on the left edge of the window

**Comment by nhcarrigan** [comment]
> Something I am not sure we entirely thought of is the tests that break when the screen is too big/small.

I had another thought on that: this is a pre-existing issue. That is, if a camper is working on the CodePen project on mobile, the suite will fail as well (if the screen is too small). It's definitely something that would be nice to fix, but I don't think it should be blocking per se.

**Comment by nhcarrigan** [comment]
Where are we at with this?

**Comment by moT01** [comment]
I think we need a meeting to catch up on some of this @nhcarrigan. We can't add this until we at least get that code saving PR in, and the code saving hotkeys PR I was working on. Both [listed under task 2 here](https://github.com/freeCodeCamp/freeCodeCamp/issues/44664). And we need to figure out how we're going to handle it - this changes the projects from codepen to our editor. We have 10 cert projects (RWD + RWD beta), do we want to change all of them to use our editor? or just the beta. Do we want this to be a new cert? or just replace the old one. We may be holding off on this until we figure out some of the versioning stuff, but I'm not sure.

**Comment by moT01** [comment]
I think this is ready for a round of reviews. It converts all 10 English RWD cert projects (RWD + RWD beta) - I believe that's the route we wanted to go. Let me know if you have any questions.

**Comment by nhcarrigan** [comment]
Haven't dug in to why, yet, but it looks like only the beta is using the multifile editor.

Non-Beta: 

image

Beta:

image

**Review by naomi-lgbt — DISMISSED** [review]
Okay, had a play with this locally, everything is looking good to me.

Was able to complete the projects, claim the cert, and view my projects + code from the cert and the settings page.

**Review by naomi-lgbt — DISMISSED** [review]
image

Still looks good to me.

**Comment by ojeytonwilliams** [comment]
We want this to be merged as part of the release, right?

I think it's ready, code-wise, but I blocked it since I figured we'd want to co-ordinate things.

File: client/src/templates/Challenges/redux/code-storage-epic.js [hunk 3]
```diff
@@ -136,9 +137,18 @@ function loadCodeEpic(action$, state$) {
       const fileKeys = challengeFiles.map(x => x.fileKey);
       const invalidForLegacy = fileKeys.length > 1;
       const { title: legacyKey } = challenge;
-
       const codeFound = getCode(id);
-      if (codeFound && isFilesAllPoly(codeFound)) {
+
+      // see if this challenge is saved in database
```
Comment (ojeytonwilliams) [inline client/src/templates/Challenges/redux/code-storage-epic.js:142]:
```suggestion
      // first check if the store (which is syncronized with the db) has saved
      // code
```
I didn't immediately get this, so an expanded comment might help.

File: client/src/templates/Challenges/redux/code-storage-epic.js [hunk 4]
```diff
@@ -136,9 +137,18 @@ function loadCodeEpic(action$, state$) {
...
+      const savedChallenges = savedChallengesSelector(state);
+      const savedChallenge = savedChallenges?.find(saved => {
+        return saved.id === challenge.id;
+      });
+
+      // skip everything if a challenge is saved in db
```
Comment (ojeytonwilliams) [inline client/src/templates/Challenges/redux/code-storage-epic.js:148]:
```suggestion
      // if the store is already populated with the saved files, we should not
      // overwrite them with the local storage data
```
It's clear from the code what's happening, but it was not obvious why we're doing that.

**Review by ojeytonwilliams — DISMISSED** [review]
LGTM :+1: 

I would comment the code slightly differently, but that's not a blocker.

**Pull request merged**